src/pluto/image.cr

```diff
@@ -33,4 +33,10 @@ abstract class Pluto::Image
   def size : Int32
     width * height
   end
+
+  private def resolve_to_start_and_count(range, size) : Tuple(Int32, Int32)
+    start, count = Indexable.range_to_index_and_count(range, size) || raise IndexError.new("Unable to resolve range #{range} for image dimension of #{size}")
```

Comment:
This is using a public _undocumented_ method from the standard library to convert a `Range` into a starting number and a count of elements. Takes care of all the math involving negative numbers, wrapping, etc., that I didn't want to reproduce in Pluto itself.